source_handlers.go

```diff
@@ -325,3 +325,37 @@ func SourceCheckAvailability(c echo.Context) error {
 
 	return c.JSON(http.StatusAccepted, map[string]interface{}{})
 }
+
+func InternalSourceList(c echo.Context) error {
+	filters, err := getFilters(c)
+	if err != nil {
+		return err
+	}
+
+	limit, offset, err := getLimitAndOffset(c)
+	if err != nil {
+		return err
+	}
+
+	// When listing sources via cert-auth we want to lock them down to only the
+	// satellite source type.
+	if c.Get("cert-auth") != nil {
+		satelliteId := strconv.Itoa(int(dao.Static.GetSourceTypeId("satellite")))
+		filters = append(filters, util.Filter{Name: "source_type_id", Value: []string{satelliteId}})
+	}
```

Comment:
This check probably isn't necessary for the internal endpoint, we can remove it!

Reply:
All right, thanks!